packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol

```diff
@@ -0,0 +1,154 @@
+  event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+  function setUp() public {
+    ph.setEpochSize(EPOCH_SIZE);
+    blockchainParameters = new BlockchainParameters(true);
+  }
+
+  function test_setBlockGasLimit_set_variable() public {
+    blockchainParameters.setBlockGasLimit(gasLimit);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+  }
+
+}
+
+contract BlockchainParametersTest_initialize is BlockchainParametersTest {
+  uint constant lookbackWindow = 20;
+
+  function test_ShouldSetTheVariables() public {
+    blockchainParameters.initialize(gasForNonGoldCurrencies, gasLimit, lookbackWindow);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+    blockTravel(EPOCH_SIZE);
+    assertEq(blockchainParameters.getUptimeLookbackWindow(), lookbackWindow);
+  }
+
+  function test_ShoudEmitIntrinsicGasForAlternativeFeeCurrencySet() public {
+    vm.expectEmit(true, true, true, true);
+    emit IntrinsicGasForAlternativeFeeCurrencySet(gasForNonGoldCurrencies);
+    emit UptimeLookbackWindowSet(lookbackWindow, 2);
+    emit OwnershipTransferred(msg.sender, msg.sender);
+    blockchainParameters.initialize(gasForNonGoldCurrencies, gasLimit, lookbackWindow);
+  }
+
+}
+
+contract BlockchainParametersTest_setBlockGasLimit is BlockchainParametersTest {
+
+  function test_ShouldSetTheVariable () public {
+    blockchainParameters.setBlockGasLimit(gasLimit);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+  }
+
+  function test_ShoudEmitBlockGasLimitSet () public {
+    vm.expectEmit(true, true, true, true);
+    emit BlockGasLimitSet(gasLimit);
+    blockchainParameters.setBlockGasLimit(gasLimit);
+  }
+
+  function test_ShouldRevert_WhenCalledByNonOwner() public {
+    vm.prank(msg.sender);
+    vm.expectRevert("Ownable: caller is not the owner");
+    blockchainParameters.setBlockGasLimit(gasLimit);
+  }
+}
+
+contract BlockchainParametersTest_setIntrinsicGasForAlternativeFeeCurrency is BlockchainParametersTest {
+  function test_ShouldSetTheVariable () public {
+    blockchainParameters.setIntrinsicGasForAlternativeFeeCurrency(gasForNonGoldCurrencies);
+    assertEq(blockchainParameters.intrinsicGasForAlternativeFeeCurrency(), gasForNonGoldCurrencies);
+  }
+
+  function test_ShoudEmitBlockGasLimitSet () public {
+    vm.expectEmit(true, true, true, true);
+    emit IntrinsicGasForAlternativeFeeCurrencySet(gasForNonGoldCurrencies);
+    blockchainParameters.setIntrinsicGasForAlternativeFeeCurrency(gasForNonGoldCurrencies);
+  }
+
+  function test_ShouldRevert_WhenCalledByNonOwner() public {
+    vm.prank(msg.sender);
+    vm.expectRevert("Ownable: caller is not the owner");
+    blockchainParameters.setIntrinsicGasForAlternativeFeeCurrency(gasForNonGoldCurrencies);
+  }
+}
+
+contract BlockchainParametersTest_getUptimeLookbackWindow is BlockchainParametersTest {
+
+  function test_ShouldRevert_WhenNotSet() public {
```

Comment:
NIT: rename test in an effort to stick to the recommended [best practices](https://book.getfoundry.sh/tutorials/best-practices#general-contract-guidance). Here and throughout